optimum/commands/export/openvino.py

```diff
@@ -281,14 +312,41 @@ def run(self):
             # avoid import when using other exporters (IPEX, INC)
             from ...exporters.openvino.convert import export_tokenizer
 
-            output = Path(self.args.output)
             tokenizer = getattr(model, "tokenizer", None)
             if tokenizer is not None:
-                export_tokenizer(tokenizer, output / "tokenizer")
+                export_tokenizer(tokenizer, self.args.output / "tokenizer")
 
             tokenizer_2 = getattr(model, "tokenizer_2", None)
             if tokenizer_2 is not None:
-                export_tokenizer(tokenizer_2, output / "tokenizer_2")
+                export_tokenizer(tokenizer_2, self.args.output / "tokenizer_2")
+        elif task.startswith("text-generation") and quantize_with_dataset:
+            from optimum.intel import OVModelForCausalLM
+
+            # To quantize a text-generation model with a dataset, an instantiated OVModelForCausalLM is required
+            model = OVModelForCausalLM.from_pretrained(
+                self.args.model,
+                export=True,
+                quantization_config=quantization_config,
+                stateful=not self.args.disable_stateful,
+                trust_remote_code=self.args.trust_remote_code,
+            )
+            model.save_pretrained(self.args.output)
+
+            tokenizer = None
+            try:
+                from transformers import AutoTokenizer
+
+                tokenizer = AutoTokenizer.from_pretrained(
+                    self.args.model, trust_remote_code=self.args.trust_remote_code
+                )
+                tokenizer.save_pretrained(self.args.output)
+            except Exception:
+                logger.warning("Could not save tokenizer")
+
+            if tokenizer and not self.args.disable_convert_tokenizer:
+                from ...exporters.openvino.convert import export_tokenizer
+
+                export_tokenizer(tokenizer, self.args.output / "tokenizer")
```

Comment:
Had to add such logic because otherwise tokenizer ends up being saved to a temporary directory.

Reply:
to simplify the code a bit what do you think about calling directly `maybe_load_preprocessors` https://github.com/huggingface/optimum-intel/blob/8dd2a895016b3a9db7a628294d13ccf1ac25a1e4/optimum/exporters/openvino/__main__.py#L344

and to save the tokenizer we can also use `maybe_save_preprocessors` https://github.com/huggingface/optimum-intel/blob/8dd2a895016b3a9db7a628294d13ccf1ac25a1e4/optimum/exporters/openvino/convert.py#L625

also we could create a function https://github.com/huggingface/optimum-intel/blob/8dd2a895016b3a9db7a628294d13ccf1ac25a1e4/optimum/exporters/openvino/__main__.py#L364-L390 that could be re-used here

Reply:
Thanks! Done